src/foundry_dev_tools/resources/dataset.py

```diff
@@ -0,0 +1,620 @@
+"""Dataset helper class."""
+
+from __future__ import annotations
+
+import io
+import os
+import pickle
+import tempfile
+import time
+from contextlib import contextmanager
+from pathlib import Path
+from typing import TYPE_CHECKING, ClassVar, overload
+
+from foundry_dev_tools.errors.compass import ResourceNotFoundError
+from foundry_dev_tools.errors.dataset import (
+    DatasetHasNoOpenTransactionError,
+)
+from foundry_dev_tools.resources import resource
+from foundry_dev_tools.utils import api_types
+
+if TYPE_CHECKING:
+    from typing import IO, AnyStr, Literal
+
+    import pandas as pd
+    import pandas.core.frame
+    import pyarrow as pa
+    import pyspark.sql
+
+    from foundry_dev_tools.config.context import FoundryContext
+
+
+class Dataset(resource.Resource):
```

Comment:
If you want to keep this in, this class needs a good end2end integration test, testing all functions.